### PR TITLE
Draw string cleanup

### PR DESCRIPTION
**Pull request by punpckhdq**
separates functions in ``render_cartographer_ingame_ui.cpp`` and prepares file for later modifications coming down the pipe

renames functions in ``draw_string.cpp`` to their real names brought in from halo 1 and creates ``rasterizer_text.cpp`` to put ``rasterizer_draw_unicode_string`` in its proper place

### CODE REVIEW

File: xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp [hunk 1]
```diff
@@ -15,44 +16,89 @@
...
-#define CARTOGRAPHER_TEST_BUILD_DRAW_TEXT
-
-#define k_status_text_font 0
+//#define ACHIVEMENT_RENDER_DEBUG_ENABLED
 
-#define k_cheevo_display_lifetime (5 * k_shell_time_msec_denominator)
-#define k_cheevo_title_font 10
-#define k_cheevo_message_font 1
+// define this to render git branch information (if GEN_GIT_VER_VERSION_STRING is defined)
+//#define CARTOGRAPHER_TEST_BUILD_DRAW_TEXT
```
Comment (Berthalamew) [inline xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: uncomment this

File: xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp [hunk 2]
```diff
@@ -198,24 +236,22 @@ void render_cartographer_update(void)
...
-		{
-			wchar_t update_message_buffer[256];
-			float percent_complate = 100.f * ((float)sizeOfDownloaded / sizeOfDownload);
-			swprintf_s(update_message_buffer, NUMBEROF(update_message_buffer), L"(progress: %.2f%%)", percent_complate);
-			draw_string_render(&bounds, update_message_buffer);
-		}
+	if (update_size_bytes > 0)
+	{
+		wchar_t update_message_buffer[256];
+		float percent_complate = 100.f * ((float)update_downloaded_bytes / update_size_bytes);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: use real32

Comment (punpckhdq) [reply xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: fixed in latest commit

File: xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp (hunk 1 above)
Comment (punpckhdq) [reply xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: fixed in latest commit

File: xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp [hunk 3]
```diff
@@ -15,44 +16,89 @@
 #include "H2MOD/Modules/Achievements/Achievements.h"
 #include "H2MOD/Modules/Shell/Config.h"
 
+#if defined(GEN_GIT_VER_VERSION_STRING)
```
Comment (Berthalamew) [inline xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: GEN_GIT_VER_VERSION_STRING is defined in `version_git.h`

**Review by Berthalamew — CHANGES_REQUESTED** [review]

File: xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp (hunk 3 above)
Comment (punpckhdq) [reply xlive/Blam/Engine/render/render_cartographer_ingame_ui.cpp]: fixed in latest commit

**Review by Berthalamew — APPROVED** [review]

**Pull request merged**